### PR TITLE
Submission flow result screens routing (EXPOSUREAPP-6034)

### PR DESCRIPTION
**Pull request by SamuraiKek**
Adds the necessary routing to display all the test results and test cards on the home screen. 
Updating the specific test result screens with the right info depending on what type of test it is, will be done in a different PR.

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/ui/testresults/negative/RATResultNegativeFragment.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import javax.inject.Inject
+
+class RATResultNegativeFragment : Fragment(R.layout.fragment_submission_antigen_test_result_negative), AutoInject {
+    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
+    private val viewModel: RATResultNegativeViewModel by cwaViewModels { viewModelFactory }
+
+    private val binding: FragmentSubmissionAntigenTestResultNegativeBinding by viewBindingLazy()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/ui/testresults/negative/RATResultNegativeFragment.kt:21]: `TODO` ?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/qrcode/scan/SubmissionQRCodeScanFragment.kt [hunk 2]
```diff
@@ -86,16 +87,57 @@ class SubmissionQRCodeScanFragment :
...
+                            )
+                        }
+                    }
+                    CoronaTestResult.PCR_NEGATIVE,
+                    CoronaTestResult.PCR_INVALID,
+                    CoronaTestResult.PCR_REDEEMED ->
+                        doNavigate(
+                            SubmissionQRCodeScanFragmentDirections
+                                .actionSubmissionQRCodeScanFragmentToSubmissionTestResultPendingFragment(
+                                    testType = CoronaTest.Type.RAPID_ANTIGEN
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/qrcode/scan/SubmissionQRCodeScanFragment.kt:121]: I think this should be `PCR` ?  ,`when` cases says `PCR_`

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/testresult/pending/SubmissionTestResultPendingFragment.kt [hunk 3]
```diff
@@ -33,9 +35,14 @@ class SubmissionTestResultPendingFragment : Fragment(R.layout.fragment_submissio
 
     private var errorDialog: AlertDialog? = null
 
+    private val navArgs by navArgs<SubmissionTestResultPendingFragmentArgs>()
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
+        pendingViewModel.updateTestType(navArgs.testType)
+        Toast.makeText(context, "Scanned ${navArgs.testType} test", Toast.LENGTH_SHORT).show()
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/testresult/pending/SubmissionTestResultPendingFragment.kt:44]: I think this should not appear to the user ?

File: Corona-Warn-App/src/main/res/layout/time_counter.xml [hunk 4]
```diff
@@ -0,0 +1,85 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/time_counter.xml:6]: Do you need the outer constraint layout?

File: Corona-Warn-App/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -1409,6 +1409,46 @@
...
+    <!-- XTXT: coronatest negative antigen result second info body -->
+    <string name="coronatest_negative_antigen_result_second_info_body">"Der Schnelltest hat keinen Nachweis für das Coronavirus SARS-CoV-2 bei Ihnen ergeben."</string>
+    <!-- XTXT: coronatest negative antigen result third info title -->
+    <string name="coronatest_negative_antigen_result_third_info_title">"Test entfernen"</string>
+    <!-- XTXT: coronatest negative antigen result third info body -->
+    <string name="coronatest_negative_antigen_restul_third_info_body">"Bitte entfernen Sie den Test wieder aus der Corona-Warn-App, damit Sie bei Bedarf einen neuen Test hinterlegen können."</string>
+
+
+
+
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/values/strings.xml:1451]: some redundant lines

File: Corona-Warn-App/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -1409,6 +1409,46 @@
...
+    <!-- XTXT: coronatest negative antigen result first info body -->
+    <string name="coronatest_negative_antigen_result_first_info_body">"Das Test-Ergebnis wird 48 Stunden hier angezeigt. Zusätzlich legt die App einen Eintrag in Ihrem Kontakt-Tagebuch an."</string>
+    <!-- XHED: coronatest negative antigen result second info title -->
+    <string name="coronatest_negative_antigen_result_second_info_title">"Befund negativ"</string>
+    <!-- XTXT: coronatest negative antigen result second info body -->
+    <string name="coronatest_negative_antigen_result_second_info_body">"Der Schnelltest hat keinen Nachweis für das Coronavirus SARS-CoV-2 bei Ihnen ergeben."</string>
+    <!-- XTXT: coronatest negative antigen result third info title -->
+    <string name="coronatest_negative_antigen_result_third_info_title">"Test entfernen"</string>
+    <!-- XTXT: coronatest negative antigen result third info body -->
+    <string name="coronatest_negative_antigen_restul_third_info_body">"Bitte entfernen Sie den Test wieder aus der Corona-Warn-App, damit Sie bei Bedarf einen neuen Test hinterlegen können."</string>
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/values/strings.xml:1447]: should those strings go to rat_strings files ?

**Review by mtwalli — COMMENTED** [review]
Nice work! , just a few small comments from my side

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/ui/testresults/negative/RATResultNegativeFragment.kt (hunk 1 above)
Comment (SamuraiKek) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/ui/testresults/negative/RATResultNegativeFragment.kt:21]: Yes

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/testresult/pending/SubmissionTestResultPendingFragment.kt (hunk 3 above)
Comment (SamuraiKek) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/testresult/pending/SubmissionTestResultPendingFragment.kt:44]: This should be removed in the end, yes. I just wanted a quick visual feedback to know that the routing works correctly when testing, as I haven't updated the UI with the right strings yet.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/qrcode/scan/SubmissionQRCodeScanFragment.kt (hunk 2 above)
Comment (SamuraiKek) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/qrcode/scan/SubmissionQRCodeScanFragment.kt:121]: Yes, thanks, I'll update it.

File: Corona-Warn-App/src/main/res/values/strings.xml (hunk 6 above)
Comment (SamuraiKek) [reply Corona-Warn-App/src/main/res/values/strings.xml:1447]: Good point, they should, and they should also go in the de strings. I wasn't aware of the rat_strings files when I first added them.

File: Corona-Warn-App/src/main/res/layout/time_counter.xml (hunk 4 above)
Comment (SamuraiKek) [reply Corona-Warn-App/src/main/res/layout/time_counter.xml:6]: Mmm, it might not, old habit I guess. I'll remove it.

**Comment by mtwalli** [comment]
Quick finding:
1- Back button in Qr-Code scan screen is not working
2- Text in Negative test card is malformed:
Screenshot 2021-04-20 at 18 33 51

**Comment by SamuraiKek** [comment]
> Quick finding:
> 1- Back button in Qr-Code scan screen is not working
> 2- Text in Negative test card is malformed:
> 3- Positive test the same:

Yes, I'll address them in another PR.

**Comment by mtwalli** [comment]
Submission flow: 

https://user-images.githubusercontent.com/25054729/115434862-7833d400-a209-11eb-80b8-f50a97c7b4f4.mp4

**Review by chiljamgossow — APPROVED** [review]
LGTM

**Comment by harambasicluka** [comment]
> Submission flow:
> 
>  device-2021-04-20-184752.mp4

I think the cards are still not wired up correctly, but afaik it's not a problem with this PR.
CC: @AlexanderAlferov & @d4rken

**Review by mtwalli — APPROVED** [review]
Great job! 👏

**Comment by d4rken** [comment]
The missing registration dates on the cards should be fixed after #2882.

**Comment by d4rken** [comment]
> Submission flow:
> 
>  device-2021-04-20-184752.mp4

That is also fixed in #2882 AFAICT, the "SubmissionDone" state was not wired up correctly.

**Review by janetback — APPROVED** [review]
UA reviewed

**Pull request merged**